### PR TITLE
Adds basic Stock Parts to Autolathe

### PR DESCRIPTION
**Pull request by uraniummeltdown**
Adds the basic versions of Stock Parts to the Autolathe under the Electronics category.
Now people can build machines without having to go through Science all the time for the components.

:cl:
Added Capacitor, Scanning Module, Micro Manipulator, Matter Bin, Micro-Laser to the Autolathe.
:cl:

### CODE REVIEW

**Comment by fludd12** [comment]
Science could just do their jobs and make the design disks for the beginner/intermediate level stock parts, at which point cargo and anyone else could upload them to the Autolathe.

**Comment by uraniummeltdown** [comment]
>Science could just do their jobs and make the design disks for the beginner/intermediate level stock parts, at which point cargo and anyone else could upload them to the Autolathe.
It's very tedious to upload the designs to disks as the disks can only hold one design each, not many scientists really know you can upload designs to disks except the experienced ones, and you sometimes get those shifts where you take a look at the RnD room and the starting metal and glass are lying there untouched.

These are only the basic parts, anyone wanting upgrades still needs to go to Science

**Comment by IcyV** [comment]
I'm not quite sure I agree with this. It even further removes any responsibility from the job and separates it even more from the station. The individual departments *should* be somewhat reliant on each other in some ways and this is one of the biggest for Science.

**Comment by Necaladun** [comment]
Agreed, it powers up anyone with access to autolathes too much.

**Comment by uraniummeltdown** [comment]
>I'm not quite sure I agree with this. It even further removes any responsibility from the job and separates it even more from the station. The individual departments should be somewhat reliant on each other in some ways and this is one of the biggest for Science.

I play Scientist/RD 90% of the time and hardly anyone ever asks for stock parts, apart from the odd Engineer who wants to build an Autolathe in Engineering or the Brig Phys asking for Body Scanner + Console parts. Science is still responsible for doing RnD and upgrading the station. Basic stock parts are not very efficient or great either.

>Agreed, it powers up anyone with access to autolathes too much.

The only Autolathes that exist at roundstart are the ones in Cargo and Science. This doesn't power up anyone with access to autolathes too much, as to build any machines they will still require a circuit board, which means breaking into / getting access to Tech Storage or asking Science for one.

Also, most of the dangerous machines (Comms Console, AI Upload Console, ID Console, Camera Monitor) aren't even machines, but computers, which can be built by any greyshirt and do not even require stock parts.

**Comment by Lady-Luck** [comment]
>Agreed, it powers up anyone with access to autolathes too much.

I do find it odd it's generally considered that science is a huge outlier in terms of job power, but PRs that try and bring just a little bit of the toys over to other jobs are generally disliked.

**Comment by IcyV** [comment]
>I do find it odd it's generally considered that science is a huge outlier in terms of job power, but PRs that try and bring just a little bit of the toys over to other jobs are generally disliked.

I'm not entirely sure how that's relevant. This isn't really much of an issue of Science being powerful or making their "toys" available to other departments. There wouldn't be too much point to Science with changes like this aside from their high-tech gadgets and xenobio. What's the point of departments if everyone can get the materials that others are supposed to provide? Research outpost mining of the past made normal mining pretty much irrelevant. This would make ever going to science for basic machines irrelevant. That and I firmly believe cargo, of all departments, should not be able to just crank out basic machines as they want provided they get the boards. People should have to work for or request these kinds of things and rely on someone else to provide it.

I don't think it's an issue of power, I think it's an issue of defeating the purpose of the job. Science is meant to research and create. Slowly opening it to anyone being able to do exactly what they can do just makes the job even more pointless than it already is.

Basically, 

>Science could just do their jobs and make the design disks for the beginner/intermediate level stock parts, at which point cargo and anyone else could upload them to the Autolathe.

**Comment by Lady-Luck** [comment]
Design disks are a gigantic pain in the ass.

**Comment by Fullofskittles** [comment]
> Design disks are a gigantic pain in the ass. This is a simpler solution.

Wouldn't the actual solution be to fix design disks then?

**Comment by Lady-Luck** [comment]
Why do we WANT Science to be the gatekeepers of the most basic machine parts?

**Comment by Fullofskittles** [comment]
> Why do we WANT Science to be the gatekeepers of the most basic machine parts?

It's their job. Jobs are meant to provide services or materials.  If the _disks_ are the problem then why not fix the _actual_ problem instead of a symptom?

> I play Scientist/RD 90% of the time and hardly anyone ever asks for stock parts

It seems that if this is the case then it's even more unnecessary.

**Comment by Fox-McCloud** [comment]
Don't agree with this, especially in combination with making R&D console's accessless.

If R&D consoles are accessless AND autolathes can print these by default, it effectively means that engineering is going to wind up with their own R&D lab---and since they were recently given access to the ore redemption machine, well...you can see where this is going.

**Comment by HereticChurch** [comment]
I like this, but I won't like it as much if the accessless R&D consoles get merged. 

People will still go to science to ask for boards if they need to build something, and may still ask R&D for stock parts anyway since they have one of the starting autolathes.

**Comment by Lady-Luck** [comment]
>since they were recently given access to the ore redemption machine, well...you can see where this is going

That actually went through?

**Comment by Lady-Luck** [comment]
I wish there was a 'thumb-sideways' reaction omji because I do think this is a good idea overall but there are some legitimate concerns here, mostly what Church said and Engineers having ore redemption permission for some reason. (Seriously why did that go through)

**Comment by uraniummeltdown** [comment]
>Don't agree with this, especially in combination with making R&D console's accessless.

>If R&D consoles are accessless AND autolathes can print these by default, it effectively means that engineering is going to wind up with their own R&D lab---and since they were recently given access to the ore redemption machine, well...you can see where this is going.

Issues with resources are a problem however I think we could solve that problem by increasing the muliplier on ore smelting in the ORM. Right now I believe with bluespace bins it multiplies ores by x2, however if we increased it to something like x3 (and increased scaling for lower tier parts) ores shouldn't be much of a problem. If a non-antag Engineer is stealing all the ores for no reason other than spamming lockboxed guns because "lol" they should probably be bwoinked for being a dick for their own amusement. However you could and still can deconstruct the ORM to drop all it's contents, which any greyshirt can do.

As for public RnD consoles, they were always possible but you had no way of knowing the feature existed unless you looked at the code.

**Comment by Lady-Luck** [comment]
>Issues with resources are a problem however I think we could solve that problem by increasing the muliplier on ore smelting in the ORM.

I don't think that's what he meant. I think he meant engineers raiding the ORM and doing science's job for them.

Honestly what needs to happen is the ORM permissions need to be reworked so that certain materials are job-restricted.

**Comment by Lady-Luck** [comment]
@uraniummeltdown I think a lot more people would be comfortable with this PR if you made it so Engineers could only get metal, glass, plasteel, and plasma-glass from the ORM, which is what that ORM restriction-PR that somehow passed should've done in the first place.

**Comment by uraniummeltdown** [comment]
>@uraniummeltdown I think a lot more people would be comfortable with this PR if you made it so Engineers could only get metal, glass, plasteel, and plasma-glass from the ORM, which is what that ORM restriction-PR that somehow passed should've done in the first place.

I'm pretty new to coding and can only really do copy-paste jobs, but if I knew how I would try and tweak the ORM like that

**Comment by GeneralChaos81** [comment]
I actually think this is pretty useful. If R&D was bombed, and the protolathe is destroyed... you'd have to cannibalize at least two other machines to just replace the protolathe. Which I guess is fine, but it'd be nice to have simple redundancy.

As a primary engineer, if this goes in, I'm going to build every machine, just because I can.

**Comment by Fox-McCloud** [comment]
>Issues with resources are a problem however I think we could solve that problem by increasing the muliplier on ore smelting in the ORM. Right now I believe with bluespace bins it multiplies ores by x2, however if we increased it to something like x3

Yeah, no, you're not powercreeping R&D back up to where it used to be.

**Comment by uraniummeltdown** [comment]
I'm not sure why people oppose this so much as it's just a little handy feature, but I guess making the Tech Storage RnD board Public means there is not so much of a need for this anymore. You can close this PR.

**Comment by monster860** [comment]
Keep in mind, it's always better to go to science than cargo.

**Comment by Fox-McCloud** [comment]
Closing on request.
